### PR TITLE
Migrate EqualPriority in scheduler to map-reduce-like framework

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #24246

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33834)

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 1]
```diff
@@ -39,51 +39,57 @@ const maxPriority float32 = 10
...
 }
 
-// CalculateSpreadPriority spreads pods across hosts and zones, considering pods belonging to the same service or replication controller.
-// When a pod is scheduled, it looks for services or RCs that match the pod, then finds existing pods that match those selectors.
-// It favors nodes that have fewer existing matching pods.
-// i.e. it pushes the scheduler towards a node where there's the smallest number of
-// pods which match the same service selectors or RC selectors as the pod being scheduled.
-// Where zone information is included on the nodes, it favors nodes in zones with fewer existing matching pods.
-func (s *SelectorSpread) CalculateSpreadPriority(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodes []*api.Node) (schedulerapi.HostPriorityList, error) {
+func getSelectors(pod *api.Pod, sl algorithm.ServiceLister, cl algorithm.ControllerLister, rsl algorithm.ReplicaSetLister) []labels.Selector {
```
Comment (gmarek) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: Please add a comment what kind of selectors we're getting here.

**Comment by gmarek** [comment]
Only nits. LGTM.

**Comment by gmarek** [comment]
@k8s-bot gke test this issue: #33617

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go (hunk 1 above)
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: done

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go [hunk 2]
```diff
@@ -39,51 +39,57 @@ const maxPriority float32 = 10
...
 	}
+	return selectors
+}
+
+func (s *SelectorSpread) getSelectors(pod *api.Pod) []labels.Selector {
+	return getSelectors(pod, s.serviceLister, s.controllerLister, s.replicaSetLister)
+}
+
+// CalculateSpreadPriority spreads pods across hosts and zones, considering pods belonging to the same service or replication controller.
+// When a pod is scheduled, it looks for services or RCs that match the pod, then finds existing pods that match those selectors.
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go]: done

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by gmarek** [comment]
LGTM

**Pull request merged**